yellowbrick/features/base.py

```diff
@@ -215,13 +225,58 @@ def fit(self, X, y=None, **kwargs):
         """
         super(DataVisualizer, self).fit(X, y, **kwargs)
 
-        # Store the classes for the legend if they're None.
-        if self.classes_ is None:
-            # TODO: Is this the most efficient method?
-            self.classes_ = [str(label) for label in np.unique(y)]
+        self._determine_target_color_type(y)
+
+        # Compute classes and colors if target type is discrete
+        if self._target_color_type == DISCRETE:
+            # Store the classes for the legend if they're None.
+            if self.classes_ is None:
+                # TODO: Is this the most efficient method?
+                self.classes_ = [str(label) for label in np.unique(y)]
+            
+# This fails in case of PCoords if subsampled target misses some values.
+            # Ensures that classes passed by user is equal to that in target
+#            if len(self.classes_)!=len(np.unique(y)):
+#                raise YellowbrickValueError("Number of unique target is not " 
+#                                            "equal to classes")
+            color_values = resolve_colors(n_colors=len(self.classes_), 
+                                          colormap=self.colormap, colors=self.color)
+            self._colors = dict(zip(self.classes_, color_values))
+            
+        # Compute target range if colors are continuous
+        elif self._target_color_type == CONTINUOUS:
+            y = np.asarray(y)
+            self.range_ = (y.min(), y.max())
```

Comment:
Awesome! We should probably add to the docstring the learned parameters of `self.range_` and `self.classes_` and when are on each based on the `_target_color_type`.

Reply:
We may also want to do a `self._colors` here as well and get the colormap for the scatterplot - does that make sense?

Reply:
Do you mean something like `self._colors=colormap`? I plan to calculate the color based on target and colormap. Then we can pass it back as `self._colors`. Can we add this functionality(to handle ranges) to `resolve_colors`!

Reply:
Yes, though the user is likely to have supplied the colormap as a string, so you would convert it to an actual colormap and normalizer via the range, e.g. 

```
self._colors = get_cmap(self.cmap)
self._norm = mpl.colors.Normalize(vmin=self.range_[0], vmax=self.range_[1])
self.cbar = mpl.colorbar.ColorbarBase(self.cax, cmap=self._colors, norm=norm)
```

This has a lot of implications though, so maybe we should hold off until we get to the scatter prototype.

Reply:
I already updated my code with this functionality. But should we find cbar here? If yes then probably we should find legends too. Or we can keep these both to be handled by the subclass of `DataViz`

Reply:
Yeh, we can leave off the `cbar` until `finalize`.

Reply:
I updated with something like this:
```python
norm = matplotlib.colors.Normalize(self.range_[0], self.range_[1])
cmap = matplotlib.cm.get_cmap(self.colormap)
color_values = cmap([norm(yi) for yi in y])
self._colors = color_values
```
It calculates color for each point. So should I let it be like this or update to return just colormap.

Reply:
Hmm, I would absolutely want to compute the color values with norm the way you did instead of leaving it to matplotlib, but I would want to do it in draw and not in fit and I wouldn't want to store the color values on the visualizer. Go ahead and just make `self._colors` the cmap for now and we'll use `self.range_` to compute the colorbar etc. in `draw` in `finalize`.